src/guide/agent-functions/index.rst

```diff
@@ -8,7 +8,7 @@ As FLAME GPU is an agent-based modelling framework, agent functions are central
 Each agent has one or more agent functions, which allow agents to interact with the environment and other agents. Additionally, agent functions can be conditionally executed based on agent states and properties.
 
 Each agent function in FLAME GPU 2 is associated with a particular agent type and is represented by an :class:`AgentFunctionDescription<flamegpu::AgentFunctionDescription>` object. This object describes the name of the function, the agent state in which it should be active, and any state transitions it should apply to the agent. The implementation code for the behaviour is written separately and is identified by using the
-:c:macro:`FLAMEGPU_AGENT_FUNCTION` macro to label the code. Alternatively, the implementations can be written in strings and loaded at runtime. The latter method is the only method supported by FLAME GPU 2's Python API.
+:c:macro:`FLAMEGPU_AGENT_FUNCTION` macro to label the code. Alternatively, the implementations can be written in strings and loaded at runtime. The latter method is supported by FLAME GPU 2's Python API which also includes a feature to permit pure Python agent functions to be specified which are transpiled to equivalent C++ at runtime.
```

Comment:
The new sentence doesn't add anything to the point that the `FALMEGPU_AGENT_FUNCTION` macro is used. 
It would be better to rewrite this from `The implementation code for the behaviour` to the end, stating that:

The implementation is specified separately, in one of 3 ways:
* As a globally scoped (CUDA) C++ function, labelled with the `FALMEGPU_AGENT_FUNCTION` macro 
* Can be written as a `FALMEGPU_AGENT_FUNCTION` labelled C++ function written as a string, in either a C++ or Python 3 model
* As a (global? file?)-scoped python 3 function annoted as `@pyflamegpu.agent_function` which are transpiled to equivalent C++ and which is compiled at runtime 
I assume there is probably a similar change required for agent function conditions, which just hasn't shown up in the review due to the current lack of support (which should be fixed).

Reply:
changed to a list format